modules/src/core/ics04_channel/handler.rs

```diff
@@ -227,3 +236,33 @@ where
     };
     Ok(())
 }
+
+fn process_write_ack<Ctx>(
+    ctx: &mut Ctx,
+    packet: Packet,
+    acknowledgement: &dyn Acknowledgement,
+    module_output: &mut ModuleOutputBuilder,
+) -> Result<(), Error>
+where
+    Ctx: Ics26Context,
+{
+    let write_ack_output =
+        write_acknowledgement::process(ctx, packet, acknowledgement.as_ref().to_vec().into())?;
+
+    let HandlerOutput {
+        result,
+        log,
+        events,
+    } = write_ack_output;
+
+    // store write ack result
+    ctx.store_packet_result(result)?;
+
+    let inner_module_output = ModuleOutputBuilder::new()
+        .with_log(log)
+        .with_events(events.into_iter().map(|event| event.into()).collect());
```

Comment:
I just realized that this conversion here from `IbcEvent` to `ModuleEvent` would not work (and should not be required). This is required now because the `packet_callback()` is restricted to add only `ModuleEvent`s to prevent modules from emitting core `IbcEvent`s. I am working on a fix for this now.

Reply:
https://github.com/informalsystems/ibc-rs/pull/2439/commits/880b5a1b7bb5c4e42cd242b40c457947fb4387b5